### DISCUSSION TITLE
What's the difference between `extend the VSCode workbench` and `Access DOM`?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [microsoft/vscode-discussions#266] [Extension Development QnA] [answered]
When I read [extension-capabilities](https://code.visualstudio.com/api/extension-capabilities/overview),

> [No DOM Access](https://code.visualstudio.com/api/extension-capabilities/overview#no-dom-access)
> Extensions have no access to the DOM of VS Code UI. You cannot write an extension that applies custom CSS to VS Code or adds an HTML element to VS Code UI.

I have a questions about it:
We know we can extend the VSCode workbench, add view to VSCode workbench, what's the difference between `extend the VSCode workbench` and `Access DOM`?

**Comment by isidorn** [comment, marked as answer]
Extending the workbench means using the VS Code API to do that. Accessing the DOM is not possible. DOM stands for Document Object Model.
